source/client-side-operations-timeout/client-side-operations-timeout.rst

```diff
@@ -0,0 +1,875 @@
+``iscryptd`` field in the server's description is ``true``.
+
+For explicit encryption and decryption, the ``ClientEncryptionOpts`` options
+type used to construct `ClientEncryption
+<../client-side-encryption/client-side-encryption.rst#clientencryption>`_
+instances MUST support a new ``timeoutMS`` option, which specifies the timeout
+for all operations executed on the ``ClientEncryption`` object.
+
+See `maxTimeMS is not added for mongocryptd`_.
+
+Background Connection Pooling
+-----------------------------
+
+Connections created as part of a connection pool’s ``minPoolSize``
+maintenance routine MUST use ``connectTimeoutMS`` as the timeout for
+connection establishment. After the connection is established, if
+``timeoutMS`` is set at the MongoClient level, it MUST be used as the timeout
+for all commands sent as part of the MongoDB or authentication handshakes.
+The timeout MUST be refreshed after each command. These commands MUST set
+timeouts per the `Command Execution`_ section. If ``timeoutMS`` is not set,
+drivers MUST continue to honor ``socketTimeoutMS`` as the socket timeout for
+handshake and authentication commands.
+
+Server Monitoring
+-----------------
+
+Drivers MUST NOT use ``timeoutMS`` for commands executed by the server
+monitoring and RTT calculation threads.
+
+See `Monitoring threads do not use timeoutMS`_.
+
+Cursors
+-------
+
+For operations that create cursors, ``timeoutMS`` can either cap the lifetime
+of the cursor or be applied separately to the original operation and all
+``next`` calls. To support both of these use cases, these operations MUST
+support a ``timeoutMode`` option. This option is an enum with possible values
+``CURSOR_LIFETIME`` and ``ITERATION``. The default value depends on the type
+of cursor being created. Drivers MUST error if ``timeoutMode`` is set and
+``timeoutMS`` is not.
+
+When applying the ``timeoutMS`` option to ``next`` calls on cursors, drivers
+MUST ensure it applies to the entire call, not individual commands. For
+drivers that send ``getMore`` requests in a loop when iterating tailable
+cursors, the timeout MUST apply to the totality of all ``getMore``’s, not to
+each one individually. If a resume is required for a ``next`` call on a
+change stream, the timeout MUST apply to the entirety of the initial
+``getMore`` and all commands sent as part of the resume attempt.
+
+For ``close`` methods, drivers MUST allow ``timeoutMS`` to be overridden if
+doing so is possible in the language. If ``timeoutMS`` is explicitly
+overridden, it MUST be honored. Otherwise, if ``timeoutMS`` was applied to
+the operation that created the cursor, drivers MUST use ``min(timeoutMS,
+connectTimeoutMS)`` as the ``timeoutMS`` value for the ``killCursors``
+command if one is required. This applies to explicit ``close`` methods that
+can be called by users as well as implicit destructors that are automatically
+invoked when exiting resource blocks.
```

Comment:
Having just read "Cursor close() methods use connectTimeoutMS as timeoutMS", I think that addresses my question above but I'd still like to confirm.

I originally read `min(timeoutMS, connectTimeoutMS)` as referring to the remaining computed timeout, but I now think that was a mistake. This is simply referring to the original `timeoutMS` option (if any) and `connectTimeoutMS`, correct?. "Server Selection" has some examples of using `min()` with `computedServerSelectionTimeout`, which would agree if so.

Reply:
A clarification wouldn't hurt and if it guarantees no one else makes the same mistake I did it will be worth it. Free to resolve this thread on your own after reading.

Reply:
Added a clarification.